dspace-spring-rest/src/main/java/org/dspace/app/rest/RestResourceController.java

```diff
@@ -336,8 +484,43 @@ public ResourceSupport listSearchMethods(@PathVariable String apiCategory, @Path
 			resources.forEach(linkService::addLinks);
 			result = assembler.toResource(resources, link);
 		} else {
-			result = repository.wrapResource((T) searchResult);
+			DSpaceResource<T> dsResource = repository.wrapResource((T) searchResult);
+			linkService.addLinks(dsResource);
+			result = dsResource;
 		}
 		return result;
 	}
+	
+	/**
+	 * Sets the location header pointing to the resource representing the given instance. Will make sure we properly
+	 * expand the URI template potentially created as self link.
+	 * 
+	 * @param headers must not be {@literal null}.
+	 * @param assembler must not be {@literal null}.
+	 * @param source must not be {@literal null}.
+	 */
+	private void addLocationHeader(HttpHeaders headers, PersistentEntityResourceAssembler assembler, Object source) {
+
+		String selfLink = assembler.getSelfLinkFor(source).getHref();
+		headers.setLocation(new UriTemplate(selfLink).expand());
+	}
+
+	@RequestMapping(method = RequestMethod.DELETE, value = "/{id:\\d+}")
+	public ResponseEntity<ResourceSupport> delete(HttpServletRequest request, @PathVariable String apiCategory,
+			@PathVariable String model, @PathVariable Integer id) throws HttpRequestMethodNotSupportedException {		
+		return deleteInternal(apiCategory, model, id);
+	}
+	
+	@RequestMapping(method = RequestMethod.DELETE, value = "/{uuid:[0-9a-fxA-FX]{8}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{4}-[0-9a-fxA-FX]{12}}")
```

Comment:
Another long regex with no comment.

Reply:
Same as above... maybe we can add define a constant UUID_REGEX, NO_UUID_OR_DIGIT_REGEX and put the comment on top of these constants